src/OrchardCore.Modules/OrchardCore.Contents/Services/DefaultContentsAdminListQueryService.cs

```diff
@@ -1,45 +1,128 @@
         {
-            // Because admin filters can add a different index to the query this must be added as a Query<ContentItem>()
+            var defaultTermNode = model.FilterResult.OfType<DefaultTermNode>().FirstOrDefault();
+            var defaultTermName = defaultTermNode?.TermName;
+            var defaultOperator = defaultTermNode?.Operation;
+            var shouldRestoreDefault = false;
+
+            if (defaultTermNode is not null)
+            {
+                var value = defaultTermNode.ToString();
+                if (!_operators.Any(opt => value.Contains(opt, StringComparison.Ordinal)))
+                {
+                    // Use an unary operator based on a full quoted string.
+                    defaultOperator = new UnaryNode(value, OperateNodeQuotes.Double);
+                }
+
+                var selectedContentType = GetSelectedContentType(model);
+
+                if (selectedContentType != null)
+                {
+                    defaultTermName = GetDefaultTermName(selectedContentType);
+                }
+
+                if (defaultTermName != defaultTermNode.TermName || defaultOperator != defaultTermNode.Operation)
+                {
+                    shouldRestoreDefault = true;
+
+                    model.FilterResult.TryRemove(defaultTermNode.TermName);
+                    model.FilterResult.TryAddOrReplace(new DefaultTermNode(defaultTermName, defaultOperator));
+                }
+            }
+
+            // Because admin filters can add a different index to the query this must be added as a Query<ContentItem>().
             var query = _session.Query<ContentItem>();
 
             query = await model.FilterResult.ExecuteAsync(new ContentQueryContext(_serviceProvider, query));
 
             // After the q=xx filters have been applied, allow the secondary filter providers to also parse other values for filtering.
-            await _contentsAdminListFilters.InvokeAsync((filter, model, query, updater) => filter.FilterAsync(model, query, updater), model, query, updater, _logger);
+            await _contentsAdminListFilters
+                .InvokeAsync((filter, model, query, updater) => filter.FilterAsync(model, query, updater), model, query, updater, _logger);
+
+            if (shouldRestoreDefault)
+            {
+                // Restore the original defaultTermNode.
+                model.FilterResult.TryRemove(defaultTermName);
+                model.FilterResult.TryAddOrReplace(defaultTermNode);
+            }
 
             return query;
         }
+
+        private static string GetSelectedContentType(ContentOptionsViewModel model)
+        {
+            if (String.IsNullOrEmpty(model.SelectedContentType))
+            {
+                var typeTermNode = model.FilterResult.OfType<ContentTypeFilterNode>().FirstOrDefault();
+
+                if (typeTermNode != null)
```

Comment:
Maybe `is not null`, and no blank line before.

Reply:
done.